module/module.go

```diff
@@ -412,17 +412,19 @@ func (m *Module) Ready(ctx context.Context, req *pb.ReadyRequest) (*pb.ReadyResp
 	}
 	m.mu.Lock()
 	defer m.mu.Unlock()
-	m.parentAddr = req.GetParentAddress()
-	if err := m.connectParent(ctx); err != nil {
-		// Return error back to parent if we cannot make a connection from module
-		// -> parent. Something is wrong in that case and the module should not be
-		// operational.
-		return nil, err
-	}
-
-	// If logger is a moduleLogger, start gRPC logging.
-	if moduleLogger, ok := m.logger.(*moduleLogger); ok {
-		moduleLogger.startLoggingViaGRPC(m)
+	if os.Getenv("VIAM_NO_MODULE_PARENT") != "true" {
```

Comment:
```suggestion
	// VIAM_NO_MODULE_PARENT=true allows a module to start without connecting to a parent module.
	if os.Getenv("VIAM_NO_MODULE_PARENT") != "true" {
```

Reply:
beyond the scope of this PR, but it might be worth discussing if this "no parent mode" is useful in other contexts / should be elevated to a setting on the module manager or something. but for now it's fine as is.